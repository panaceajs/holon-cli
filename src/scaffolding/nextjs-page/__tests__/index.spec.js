const { resolve } = require('path');
const snapshot = require('../../../utils/task-snapshot');
const task = require('../');

describe('actions', () => {
  it('should create action types in `actions/index.js`', async () => {
    await expect(
      snapshot(task, {
        create: true,
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });

  it('should add import action types statmenett`', async () => {
    await expect(
      snapshot(task, {
        cwd: resolve(__dirname, '__fixtures__/without-action-type-import'),
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });
  it('should add import action types statmenett`', async () => {
    await expect(
      snapshot(task, {
        cwd: resolve(__dirname, '__fixtures__/with-action-type-import'),
        actionTypes: { existingAction: 'existingAction' }
      })
    ).resolves.toMatchSnapshot();
  });
});
