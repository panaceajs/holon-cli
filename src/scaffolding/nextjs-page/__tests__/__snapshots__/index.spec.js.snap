// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should add import action types statmenett\` 1`] = `
"
--------------------------------------------------------------------------------
// ./store/reducer.js
// Create
--------------------------------------------------------------------------------

import { AN_ACTION, ANOTHER_ACTION } from '../action-types';

export default (
  state = {
    anAction: 'anAction-initial-value',
    anotherAction: 'anotherAction-initial-value'
  },
  action
) => {
  const { type, payload, error } = action;

  switch (type) {
    case AN_ACTION: {
      return {
        ...state,
        anAction: payload || 'anAction-value',
        error
      };
    }

    case ANOTHER_ACTION: {
      return {
        ...state,
        anotherAction: payload || 'anotherAction-value',
        error
      };
    }

    default:
      return state;
  }
};



"
`;

exports[`actions should add import action types statmenett\` 2`] = `
"
--------------------------------------------------------------------------------
// ./store/reducer.js
// Create
--------------------------------------------------------------------------------

import { EXISTING_ACTION } from '../action-types';

export default (
  state = {
    existingAction: 'existingAction-initial-value'
  },
  action
) => {
  const { type, payload, error } = action;

  switch (type) {
    case EXISTING_ACTION: {
      return {
        ...state,
        existingAction: payload || 'existingAction-value',
        error
      };
    }

    default:
      return state;
  }
};



"
`;

exports[`actions should create action types in \`actions/index.js\` 1`] = `
"
--------------------------------------------------------------------------------
// ./store/reducer.js
// Create
--------------------------------------------------------------------------------

import { AN_ACTION, ANOTHER_ACTION } from '../action-types';

export default (
  state = {
    anAction: 'anAction-initial-value',
    anotherAction: 'anotherAction-initial-value'
  },
  action
) => {
  const { type, payload, error } = action;

  switch (type) {
    case AN_ACTION: {
      return {
        ...state,
        anAction: payload || 'anAction-value',
        error
      };
    }

    case ANOTHER_ACTION: {
      return {
        ...state,
        anotherAction: payload || 'anotherAction-value',
        error
      };
    }

    default:
      return state;
  }
};



"
`;
